packages/pages/src/common/src/function/internal/types.ts

```diff
@@ -0,0 +1,116 @@
+import { ServerlessFunction, FunctionModule } from "../types.js";
+import {
+  validateFunctionDefaultExport,
+  validateFunctionModule,
+} from "./validateFunctionModule.js";
+import { convertToPluginEvent, PluginEvent } from "../../ci/ci.js";
```

Comment:
If you move these types to here then ci.js doesn't need them at all. The internal type should already have the converted values by the time it needs to dump to the ci.json file.

Reply:
I moved the convert function to internal/types

ci/ci.ts defines the types used in ci.json so I think PluginEvent still makes sense there
generate/ci.ts is where ci.json is actually written